src/codemagic/tools/app_store_connect.py

```diff
@@ -127,13 +133,20 @@ def _get_resource(self, resource_id, resource_manager, should_print):
         self.printer.print_resource(resource, should_print)
         return resource
 
-    def _list_resources(self, resource_filter, resource_manager, should_print):
+    def _list_resources(self, resource_filter, resource_manager, should_print, application_id=None):
         try:
-            resources = resource_manager.list(resource_filter=resource_filter)
+            if isinstance(resource_manager, type(self.api_client.app_store_versions)):
+                resources = resource_manager.list(application_id, resource_filter)
+            else:
+                resources = resource_manager.list(resource_filter=resource_filter)
         except AppStoreConnectApiError as api_error:
             raise AppStoreConnectError(str(api_error))
 
-        self.printer.log_found(resource_manager.resource_type, resources, resource_filter)
+        self.printer.log_found(
+            resource_manager.resource_type,
+            resources if isinstance(resources, List) else resources[0],
```

Comment:
It is not transparent at all why something that isn't list is accessed by numerical index. Also, `isinstance` check should be done with `list` type, and not with `List` type annotation type.

Reply:
not relevant anymore https://github.com/codemagic-ci-cd/cli-tools/pull/67/commits/87fa4db77f534bc9430952896e383ea509531914#diff-c655fbe88a18ddd4d367405c06673842a69855a44563b37bd3b06625bf712419R137